### PR TITLE
OpenVINO's Inference Engine as optional dependency.

### PR DESCRIPTION
**Pull request by jpablomch**
This PR adds OpenVINO as optional dependency of Scanner.

### CODE REVIEW

File: deps.sh [hunk 1]
```diff
@@ -385,6 +398,38 @@ elif [[ $INSTALL_ALL == false ]]; then
     fi
 
     echo "Optional dependencies: "
+    if [[ -z ${WITH_OPENVINO+x} ]] && [[ ${NO_OPENVINO+x} != true ]] && [[ "$OSTYPE" == "linux-gnu" ]]; then
+        echo -n "Do you need support for OpenVino Inference Engine? [Y/n]: "
```
Comment (apoms) [inline deps.sh]: Can we switch this to 'y/N' as a default for now? I suspect, at least initially, most users won't require this as a dependency when installing from scratch

File: deps.sh [hunk 2]
```diff
@@ -555,28 +639,21 @@ if [[ $INSTALL_OPENCV == true ]] && [[ ! -f $BUILD_DIR/opencv.done ]]; then
               -D BUILD_opencv_xfeatures2d=OFF \
               -D WITH_PROTOBUF=OFF \
               -D BUILD_PROTOBUF=OFF \
-              -D BUILD_opencv_dnn=OFF \
               -D OPENCV_EXTRA_MODULES_PATH=$BUILD_DIR/opencv_contrib/modules \
-              $(echo $CMDS) -DCMAKE_PREFIX_PATH=$(echo $PY_EXTRA_CMDS) \
-              .. && \
+              -D WITH_INF_ENGINE=ON \
```
Comment (apoms) [inline deps.sh]: What does this flag do?

File: deps.sh [hunk 3]
```diff
@@ -555,28 +639,21 @@ if [[ $INSTALL_OPENCV == true ]] && [[ ! -f $BUILD_DIR/opencv.done ]]; then
...
+              -D ENABLE_CXX11=ON \
+              -D BUILD_opencv_dnn=ON \
+              -D OPENCV_ENABLE_NONFREE=ON \
+              -D WITH_PROTOBUF=ON \
+              -D BUILD_PROTOBUF=ON \
+              -D BUILD_LIBPROTOBUF_FROM_SOURCES=OFF \
+              -D ENABLE_PRECOMPILED_HEADERS=OFF \
```
Comment (apoms) [inline deps.sh]: Why is PRECOMPILED_HEADERS=OFF required?

File: deps.sh (hunk 2 above)
Comment (jpablomch) [reply deps.sh]: It allows OpenCV to use OpenVINO's Inference Engine if it is available.

File: deps.sh (hunk 3 above)
Comment (jpablomch) [reply deps.sh]: I removed it. Thanks.

**Pull request merged**